cpp/include/cudf/io/writers.hpp

```diff
@@ -282,7 +282,7 @@ struct writer_options {
 
   // string to use for null entries:
   //
-  std::string const na_rep_;
+  std::string na_rep_;
```

Comment:
Why were the `const` removed?

Reply:
To use `write_csv_args` in cython it should have default constructor and we can't have a default constructor with `std::string const` so I had to remove them.

Reply:
```
cudf/_lib/csv.cpp: In function ‘PyObject* __pyx_f_4cudf_4_lib_3csv_write_csv(__pyx_obj_4cudf_4_lib_5table_Table*, int, __pyx_opt_args_4cudf_4_lib_3csv_write_csv*)’:
cudf/_lib/csv.cpp:7421:264: error: use of deleted function ‘cudf::experimental::io::write_csv_args& cudf::experimental::io::write_csv_args::operator=(cudf::experimental::io::write_csv_args&&)’
 write_csv_args(__pyx_v_snk, __pyx_v_input_table_view, __pyx_v_na_c, __pyx_v_include_header_c, __pyx_v_rows_per_hunk_c, __pyx_v_line_term_c, __pyx_v_delim_c, __pyx_v_true_value_c, __pyx_v_false_value_c, (&__pyx_v_metadata_));
                                                                                                                                                                                                                               ^
In file included from cudf/_lib/csv.cpp:747:0:
../../cpp/include/cudf/io/functions.hpp:280:8: note: ‘cudf::experimental::io::write_csv_args& cudf::experimental::io::write_csv_args::operator=(cudf::experimental::io::write_csv_args&&)’ is implicitly deleted because the default definition would be ill-formed:
 struct write_csv_args : detail::csv::writer_options {
        ^~~~~~~~~~~~~~
../../cpp/include/cudf/io/functions.hpp:280:8: error: use of deleted function ‘cudf::experimental::io::detail::csv::writer_options& cudf::experimental::io::detail::csv::writer_options::operator=(const cudf::experimental::io::detail::csv::writer_options&)’
In file included from ../../cpp/include/cudf/io/functions.hpp:26:0,
                 from cudf/_lib/csv.cpp:747:
../../cpp/include/cudf/io/writers.hpp:235:8: note: ‘cudf::experimental::io::detail::csv::writer_options& cudf::experimental::io::detail::csv::writer_options::operator=(const cudf::experimental::io::detail::csv::writer_options&)’ is implicitly deleted because the default definition would be ill-formed:
 struct writer_options {
        ^~~~~~~~~~~~~~
../../cpp/include/cudf/io/writers.hpp:235:8: error: passing ‘const string {aka const std::__cxx11::basic_string<char>}’ as ‘this’ argument discards qualifiers [-fpermissive]
In file included from /usr/include/c++/7/string:52:0,
                 from /usr/include/c++/7/bits/locale_classes.h:40,
                 from /usr/include/c++/7/bits/ios_base.h:41,
                 from /usr/include/c++/7/ios:42,
                 from cudf/_lib/csv.cpp:686:

```

Reply:
Why does it need a default ctor?

Reply:
Because Cython doesn't have brace initialization the way C++ does. You can't do in Cython:

```
cdef Foo f{x, y}
```

You must do:

```
cdef Foo f = Foo(x, y)
```

If the C++ change is too intrusive, we can use a heap allocated object instead:

```
cdef Foo* f = new Foo(x, y)
```

Reply:
Or I guess wrap it in a `unique_ptr`:

```
cdef unique_ptr[Foo] f = make_unique[Foo](x, y)
```

Reply:
@shwina  But, if we are passing `Foo` as argument and not `unique_ptr` of it, will it not cause same issue ?

Reply:
@shwina Thank you, `unique_ptr` worked.

Reply:
@shwina does it really have to do with brace initialization. Can't you just do:
```
cdef Foo f(x, y); // parenthesis instead of braces
```